parsec/core/cli/rsync.py

```diff
@@ -21,22 +21,20 @@ async def _import_file(workspace_fs: WorkspaceFS, src: FsPath, dest: FsPath):
     try:
         await workspace_fs.touch(dest)
     except FileExistsError:
-        await workspace_fs.truncate(dest, 0)
-    finally:
-        for offset, chunk in _chunks_from_path(workspace_fs, src):
-            await workspace_fs.write_bytes(dest, chunk, offset)
+        pass
+    f = await workspace_fs.open_file(dest, "w")
+    await f.write(b"".join(_chunks_from_path(workspace_fs, src)))
+    await f.close()
```

Comment:
Two comments:
- There's an async context manager for workspace files to make sure the file is closed
- Joining the chunks for a single write might fill up the RAM, it's better to call `f.write` repeatedely
```python
async with workspace_fs.open_file(dest, "w") as dest_f:
    for chunk in _chunks_from_path(workspace_fs, src):
        await dest_f.write(chunk)
```